### PR TITLE
Dev/normalization/sysmon data types

### PR DESCRIPTION
**Pull request by YuvalNaor**
Fixes #

## Proposed Changes

  -
  -
  -

### CODE REVIEW

File: Parsers/ASimNetworkSession/ProductParsers/vimNetworkSessionMicrosoft365Defender.yaml [hunk 1]
```diff
@@ -83,9 +83,9 @@ ParserQuery: |
        SrcUsername = User,
        SrcDvcId = DvcId,
        SrcDvcIdType = 'MDEid',
-       SrcUserIdType = "SID"
+       SrcUserIdType = "SID",
+      DstHostname = tostring(UrlHostname)
```
Comment (oshezaf) [inline Parsers/ASimNetworkSession/ProductParsers/vimNetworkSessionMicrosoft365Defender.yaml:87]: I wonder. How can UrlHostname not be a string?

File: Parsers/ASimNetworkSession/ProductParsers/vimNetworkSessionMicrosoft365Defender.yaml [hunk 2]
```diff
@@ -141,7 +141,7 @@ ParserQuery: |
        SrcDomain = UrlDomain,
        SrcFQDN = RemoteUrl
     | extend  
-       DstHostname = DvcHostname,
+       DstHostname = tostring(DvcHostname),
```
Comment (oshezaf) [inline Parsers/ASimNetworkSession/ProductParsers/vimNetworkSessionMicrosoft365Defender.yaml:144]: Again, did you see an example of DvcHostname not being a string?

File: Parsers/ASimNetworkSession/ProductParsers/vimNetworkSessionMicrosoftWindowsEventFirewall.yaml [hunk 3]
```diff
@@ -68,45 +68,46 @@ ParserQuery: |
...
-            | parse EventData with * '<Data Name="ProcessId">'ProcessId'</Data>'
+            | parse EventData with * '<Data Name="ProcessId">'ProcessId:string'</Data>'
                                     '\x0d\x0a  <Data Name="Application">'Application'</Data>'
                                     '\x0d\x0a  <Data Name="Direction">'DirectionCode'</Data>'
                                     '\x0d\x0a  <Data Name="SourceAddress">'SourceAddress'</Data>'
                                     '\x0d\x0a  <Data Name="SourcePort">'SourcePort'</Data>'
                                     '\x0d\x0a  <Data Name="DestAddress">'DestAddress'</Data>'
                                     '\x0d\x0a  <Data Name="DestPort">'DestPort'</Data>'
-                                    '\x0d\x0a  <Data Name="Protocol">'Protocol'</Data>'
+                                    '\x0d\x0a  <Data Name="Protocol:int">'Protocol'</Data>'
```
Comment (oshezaf) [inline Parsers/ASimNetworkSession/ProductParsers/vimNetworkSessionMicrosoftWindowsEventFirewall.yaml:78]: Mistake here

**Review by oshezaf — CHANGES_REQUESTED** [review]

File: Parsers/ASimNetworkSession/ProductParsers/vimNetworkSessionMicrosoft365Defender.yaml (hunk 1 above)
Comment (YuvalNaor) [reply Parsers/ASimNetworkSession/ProductParsers/vimNetworkSessionMicrosoft365Defender.yaml:87]: sometimes after parsing some fields remain "dynamic" and to "string", so this is for extra caution

File: Parsers/ASimNetworkSession/ProductParsers/vimNetworkSessionMicrosoft365Defender.yaml (hunk 2 above)
Comment (YuvalNaor) [reply Parsers/ASimNetworkSession/ProductParsers/vimNetworkSessionMicrosoft365Defender.yaml:144]: yup, dynamic type

**Comment by YuvalNaor** [comment]
@oshezaf fixed

File: Parsers/ASimNetworkSession/ProductParsers/vimNetworkSessionMicrosoftWindowsEventFirewall.yaml [hunk 4]
```diff
@@ -68,45 +68,46 @@ ParserQuery: |
...
-            | parse EventData with * '<Data Name="ProcessID">'ProcessID'</Data>'
+            | parse EventData with * '<Data Name="ProcessID">'ProcessID:string'</Data>'
                                     '\x0d\x0a  <Data Name="Application">'Application'</Data>'
                                     '\x0d\x0a  <Data Name="Direction">'DirectionCode'</Data>'
                                     '\x0d\x0a  <Data Name="SourceAddress">'SourceAddress'</Data>'
                                     '\x0d\x0a  <Data Name="SourcePort">'SourcePort'</Data>'
                                     '\x0d\x0a  <Data Name="DestAddress">'DestAddress'</Data>'
                                     '\x0d\x0a  <Data Name="DestPort">'DestPort'</Data>'
-                                    '\x0d\x0a  <Data Name="Protocol">'Protocol:int'</Data>'
+                                    '\x0d\x0a  <Data Name="Protocol:int">'Protocol:int'</Data>'
```
Comment (oshezaf) [inline Parsers/ASimNetworkSession/ProductParsers/vimNetworkSessionMicrosoftWindowsEventFirewall.yaml:104]: At least on my side this one still looks incorrect.

**Review by oshezaf — CHANGES_REQUESTED** [review]
One small issue remains.

**Review by oshezaf — APPROVED** [review]

**Pull request merged**